src/shallow/Snapshot/Serialize/IndentedOutput.lua

```diff
@@ -0,0 +1,60 @@
+local IndentedOutput = {}
+local IndentedOutputMetatable = {
+	__index = IndentedOutput,
+}
+
+function IndentedOutput.new(indentation, removeTrailingWhitespaces)
+	indentation = indentation or 2
+	if removeTrailingWhitespaces == nil then
+		removeTrailingWhitespaces = true
+	end
+
+	local output = {
+		_level = 0,
+		_indentation = (" "):rep(indentation),
+		_lines = {},
+		_removeTrailingWhitespaces = removeTrailingWhitespaces,
+	}
+
+	setmetatable(output, IndentedOutputMetatable)
+
+	return output
+end
+
+function IndentedOutput:write(line, ...)
+	if select("#", ...) > 0 then
+		line = line:format(...)
+	end
+
+	if self._removeTrailingWhitespaces and line == "" then
+		table.insert(self._lines, line)
```

Comment:
Yes you made me realize it should not be called that I think. Because I am not sure it's worth checking for all trailing whitespaces, instead we can just assume with not going to write a line that ends with spaces.

But the thing that the consumer of IndentedOutput does not control is when it writes an empty line in order to skip a line. Then if there is an indentation level, I don't want to indent the empty string with spaces, I just want to add it so it creates a new line when concatenated.

Reply:
Yeah that's definitely something else altogether.

Is this just for adding spacing between the requires and the returned table? It seems like a pretty specific edge case, but I think it might be reasonable to _always_ avoid indenting empty lines.

Reply:
Yes otherwise it's a bit annoying when you open the snapshot file with an editor, it removes the indentation from the empty line so then if it's already committed it appears in the diff. It's a bit of 
 a detail but I think it's worth it